scst/src/scst_targ.c

```diff
@@ -3875,6 +3871,19 @@ static int scst_finish_cmd(struct scst_cmd *cmd)
 		}
 	}
 
+	if (likely(cmd->tgt_dev != NULL)) {
+		/*
+		 * We must decrement @tgt_dev->tgt_dev_cmd_count
+		 * after scst_tgt_cmd_done() was called.
+		 *
+		 * @cmd processing for target device is complete.
+		 */
+		smp_mb__before_atomic_dec();
+		WARN_ON_ONCE(!cmd->owns_refcnt);
+		cmd->owns_refcnt = false;
```

Comment:
This assignment should not be moved since `percpu_ref_put(&cmd->dev->refcnt);` has not been moved.